app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,11 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @single_user = User.find(params[:user_id].to_i)
+    @user_posts = Post.where(author: @single_user)
+  end
 
-  def show; end
+  def show
+    @single_post = Post.find(params[:id].to_i)
+    @post_comments = Comment.where(post: @single_post)
+  end
```

Comment:
- [ ] Attempt to show single or all posts by a user throws error, like mentioned in the previous review. Kindly ensure to the show action is directed to the appropriate view template.

**View post error:**
![image](https://user-images.githubusercontent.com/12745474/215151966-23a6cf60-415b-4234-9375-504c7f1c249f.png)